### DISCUSSION TITLE
Pinia 2.0.21 introduces infinite recursion

### DISCUSSION THREAD

**Discussion by rien** [vuejs/pinia#1756] [Help and Questions] [answered]
I'm using Vue 2.7.13 with webpack.

Updating from pinia 2.0.20 to 2.0.21 causes a page to hang and eventually crash with `too many recursions` in Vue's `traverse` function: https://github.com/vuejs/vue/blob/main/src/core/observer/traverse.ts

I have no idea how this update could break this, when [comparing](https://github.com/vuejs/pinia/compare/pinia@2.0.20...pinia@2.0.21) the changes it seems that commit https://github.com/vuejs/pinia/commit/6efa780f2c6e401e1d82a80a2aeceeac962f1c4e removing a browser might be the only thing that could have his effect.

There was recently a PR in Vue 2.7 fixing a bug in this function https://github.com/vuejs/vue/pull/12806, I guess the issue we're having might be related?

Any ideas what could cause and/or fix this?

**Comment by rien** [comment]
It seems to be accessing a `_p` Object (with `install` and `state` properties) while traversing, this seems to be the `Pinia` object itself?

**Reply by posva** [reply]
yes, it is. It's marked as raw though to tell Vue not to traverse it... I will have to check if I can reproduce it with https://github.com/piniajs/example-vue-2-vite

**Reply by posva** [reply]
I couldn't reproduce it so if anybody could reproduce it at https://github.com/piniajs/example-vue-2-vite that would help

**Reply by rien** [reply]
Maybe the fact that we're using webpack (5) is important here?

Do you have any idea what could be causing this in the [changes between 2.0.20 and 2.0.21](https://github.com/vuejs/pinia/compare/pinia@2.0.20...pinia@2.0.21)? I'm guessing the browser export, but I'm not familiar enough with these kind of internals to know why.

I'll try reproducing once we have a bit more bandwidth.

**Reply by posva** [reply]
yeah, you might need to configure webpack to explicitely set the imported file for pinia

**Comment by rien** [comment, marked as answer]
It seems to be fixed now, no idea what caused it and how it's fixed now.
